src/nl/rubensten/texifyidea/inspections/TexifyRegexInspection.kt

```diff
@@ -172,31 +239,72 @@ abstract class TexifyRegexInspection(
         return mathMode == element.inMathContext() && checkContext(element)
     }
 
+    override fun runForWholeFile(): Boolean {
+        // We assume the quickfix of this inspection replaces text (like <<) by
+        // content with a different length (like \ll), so we have to make
+        // sure it is run for the whole file at once.
+        // It can be disabled by overriding this method.
+        return true
+    }
+
     /**
      * Replaces all text in the replacementRange by the correct replacement.
+     *
+     * @return The total increase in document length, e.g. if << is replaced by \ll and \usepackage{amsmath} is added then the total increase is 3 + 20 - 2.
      */
-    open fun applyFix(project: Project, descriptor: ProblemDescriptor, replacementRange: IntRange, replacement: String, groups: List<String>) {
+    open fun applyFix(project: Project, descriptor: ProblemDescriptor, replacementRange: IntRange, replacement: String, groups: List<String>): Int {
         val file = descriptor.psiElement as PsiFile
-        val document = file.document() ?: return
+        val document = file.document() ?: return 0
 
         document.replaceString(replacementRange.start, replacementRange.endInclusive, replacement)
+
+        return replacement.length - (replacementRange.endInclusive - replacementRange.start)
+    }
+
+    /**
+     * Replaces all text for all replacementRanges by the correct replacements.
+     *
+     * We assume the quickfix of an inspection performs replacements which could
+     * have a different length than the original text, the inspection
+     * should override [runForWholeFile] to explicitly refute this assumption.
+     *
+     * @param groups Regex groups as matched by the regex matcher.
+     */
+    open fun applyFixes(project: Project, descriptor: ProblemDescriptor, replacementRanges: List<IntRange>, replacements: List<String>, groups: List<List<String>>) {
```

Comment:
```suggestion
    open fun applyFixes(
            project: Project,
            descriptor: ProblemDescriptor,
            replacementRanges: List<IntRange>,
            replacements: List<String>,
            groups: List<List<String>>
    ) {
```
I do tend to be very lenient when it comes to line length with function signatures, but this is slightly too long for my taste.